### PR TITLE
Close: Decoupled state transition and the handling of the request.

### PR DESCRIPTION
**Pull request by j2gg0s**
Decoupled state transition and the handling of the request.
PR for [issue #13 ](https://github.com/eleme/doctor/issues/13)

### CODE REVIEW

File: doctor/checker.py [hunk 1]
```diff
@@ -134,56 +137,42 @@ def test(self, service_name, func_name, logger=None):
...
-                    else:
-                        # not lucky
-                        result = False
-            else:
+            if (self._metrics.api_latest_state.get(key, False) and
+                    locked_span >= self._max_recovery_time):
+                # unlock api
+                lock['locked_at'] = 0
+                lock['locked_status'] = lock_changed = MODE_UNLOCKED
+            elif not self._metrics.api_latest_state.get(key, False):
```
Comment (damnever) [inline doctor/checker.py]: Typing a few more lines of code could be better :-）
```Python
if api_latest_state_ok():
    if time_passed:
        pass
else:
    pass
```

```Python
latest_state_ok = api_latest_state_ok()
if latest_state and time_passed:
    pass
else not latest_state:
    pass

File: doctor/checker.py [hunk 2]
```diff
@@ -134,56 +137,42 @@ def test(self, service_name, func_name, logger=None):
...
-            if not health_ok_now:
-                # turns BAD
-                lock['locked_at'] = time_now
-                lock['locked_status'] = MODE_LOCKED
-                lock_changed = MODE_LOCKED
-                result = False
-            else:
-                # still OK
+                lock['locked_status'] = lock_changed = MODE_LOCKED
+        elif locked_status == MODE_UNLOCKED and \
```
Comment (damnever) [inline doctor/checker.py]: https://www.python.org/dev/peps/pep-0008/#multiline-if-statements

**Review by damnever — CHANGES_REQUESTED** [review]

**Comment by damnever** [comment]
LGTM

File: doctor/checker.py (hunk 1 above)
Comment (JustScrapy) [inline doctor/checker.py]: +1

**Review by damnever — APPROVED** [review]

**Pull request closed without merging**